edition/voxel_tool_terrain.cpp

```diff
@@ -110,7 +111,7 @@ Ref<VoxelRaycastResult> VoxelToolTerrain::raycast(
 		if (library_ref.is_null()) {
 			return res;
 		}
-		RaycastPredicateBlocky predicate{ _terrain->get_storage(), library_ref->get_baked_data(), p_collision_mask };
+		RaycastPredicateBlocky predicate{ _terrain->get_storage(), library_ref->get_baked_data(), p_collision_mask, p_pos, p_pos + p_dir * max_distance};
```

Comment:
The raycast takes place in terrain space, however you are sending global space coordinates. Use `local_pos` and `local_dir` instead.

Reply:
Fixed